File: Otel_Rezervasyon_Api/models/Kullanici.js
const mongoose = require('mongoose');

const kullaniciSchema = new mongoose.Schema({
    ad: String,
    soyadi: String,
    eposta: { type: String, required: true, unique: true },
    sifre: { type: String, required: true },
    rezervasyon_list: { type: [mongoose.SchemaTypes.ObjectId], ref: 'Rezervasyon' },
    favori_list: { type: [mongoose.SchemaTypes.ObjectId], ref: 'Otel' },
    dosya_src: String,
})

module.exports = mongoose.model('Kullanici', kullaniciSchema);

File: otel_rezervasyon/lib/models/Kullanici.dart
import 'package:otel_rezervasyon/models/Otel.dart';

class Kullanici {
  Kullanici({
    this.id,
    this.ad,
    this.eposta,
    this.sifre,
    this.rezervasyonList,
    this.favoriList,
  });

  String? id;
  String? ad;
  String? eposta;
  String? sifre;
  dynamic rezervasyonList;
  dynamic favoriList;

  factory Kullanici.fromJson(Map<String, dynamic> json) => Kullanici(
        id: json["_id"],
        ad: json["ad"],
        eposta: json["eposta"],
        sifre: json["sifre"],
        rezervasyonList: json["rezervasyon_list"],
        favoriList: json["favori_list"],
      );

  Map<String, dynamic> toJson() => {
        "ad": ad ?? '',
        "eposta": eposta ?? '',
        "sifre": sifre ?? '',
      };
}
